src/misc/ERC20.sol

```diff
@@ -49,35 +43,38 @@ contract ERC20 is Auth, IERC20Metadata, IERC20Permit {
         _DOMAIN_SEPARATOR = EIP712Lib.calculateDomainSeparator(nameHash, versionHash);
     }
 
-    function _balanceOf(address user) internal view virtual returns (uint256) {
-        return balances[user];
+    //----------------------------------------------------------------------------------------------
+    // Administration
+    //----------------------------------------------------------------------------------------------
+
+    function file(bytes32 what, string memory data) public virtual auth {
+        if (what == "name") name = data;
+        else if (what == "symbol") symbol = data;
+        else revert FileUnrecognizedParam();
+        emit File(what, data);
     }
 
+    //----------------------------------------------------------------------------------------------
+    // ERC20 balances
+    //----------------------------------------------------------------------------------------------
+
     /// @inheritdoc IERC20
     function balanceOf(address user) public view virtual returns (uint256) {
         return _balanceOf(user);
     }
 
-    function _setBalance(address user, uint256 value) internal virtual {
-        balances[user] = value;
+    function _balanceOf(address user) internal view virtual returns (uint256) {
+        return balances[user];
```

Comment:
Both are exactly the same, maybe the internal one is not need it?

Reply:
It is because it's `virtual` and overriden in `ShareToken.sol`

Reply:
Curious, why `_balanceOf()` is not public in `ShareToken.sol`?

Reply:
Since users can use `balanceOf()` for the same purpose